public/app/features/dashboard/state/utils.ts

```diff
@@ -0,0 +1,12 @@
+import { PanelModel } from './PanelModel';
+
+export function isOnTheSameGridRow(sourcePanel: PanelModel, otherPanel: PanelModel): boolean {
+  if (
+    otherPanel.gridPos.x >= sourcePanel.gridPos.x + sourcePanel.gridPos.w &&
+    otherPanel.gridPos.y === sourcePanel.gridPos.y
+  ) {
+    return true;
+  }
+
+  return false;
```

Comment:
feels like we are missing something here, for example what happens if the repeated panel is after a non repeated panel (on the same row). Wonder if we can just check is the y is the same? The other thing that could be missing is if the repeat direction is vertical, then we should push it down no matter what

Reply:
> Feels like we are missing something here, for example what happens if the repeated panel is after a non repeated panel (on the same row). Wonder if we can just check is the y is the same?

The tests I've added seems to pass with repeated panels after non repeating panel.

> The other thing that could be missing is if the repeat direction is horizontal, then we should push it down no matter what

I'll have to look into this further

Reply:
> feels like we are missing something here, for example what happens if the repeated panel is after a non repeated panel (on the same row). 

Think I know why we do not need to think about this case as that penel (before) will not be included in the for loop as we start at panelIndex and only look at panels past it (index wise), so if the panels are sorted by x & y then there should not be a panel with a lower x  passed to this function